Cargo.toml

```diff
@@ -24,11 +24,12 @@ hmac = "0.12.1"
 serde = { version = "1.0.136", features = ["derive"] }
 serde_json = "1.0.79"
 sha2 = "0.10.2"
-tokio = { version = "1.17.0", features = ["full"] }
+tokio = { version = "1.17.0", features = ["rt-multi-thread", "io-util", "macros", "net", "time"] }
 tracing = "0.1.32"
 tracing-subscriber = "0.3.10"
 uuid = { version = "0.8.2", features = ["serde", "v4"] }
 
 [dev-dependencies]
 lazy_static = "1.4.0"
 rstest = "0.12.0"
+tokio = { version = "1.17.0", features = ["sync"] }
```

Comment:
Does this actually split up the feature properly so that it's only used in `cargo test`? Just curious, I haven't seen features split between different targets before.

Reply:
Yup! It's a pretty neat feature of Cargo. Since Cargo takes the union of the features, the `test` configuration should pick up all the specified features. But, if we're only running `cargo build` (i.e. some non-`test` configuration), it _should_ ignore the `sync` feature. But if it's already in cache (i.e. somewhere in the `target` folder), anyway, then I believe it would be included.